packages/catalog-client/src/types.ts

```diff
@@ -53,3 +53,17 @@ export type AddLocationResponse = {
 export type DiscoveryApi = {
   getBaseUrl(pluginId: string): Promise<string>;
 };
+/**
+ * This is a copy of the core IdentityApi, to avoid importing core.
+ */
+export type ProfileInfo = {
+  email?: string;
+  displayName?: string;
+  picture?: string;
+};
+export type IdentityApi = {
+  getUserId(): string;
+  getProfile(): ProfileInfo;
+  getIdToken(): Promise<string | undefined>;
+  signOut(): Promise<void>;
```

Comment:
I think, this  method is unnecessary.

Reply:
Unnecessary as part of the IdentityApi interface (I haven't modified it) or unnecessary in this copy of the type, only used to avoid importing core?

Reply:
I think IdentityApi should not have a signOut method, but only the SessionAPI should . But that is not in scope of this PR as such.

Nevertheless, the Catalog Client needs to take a dependency that has a method that provides it with a token. And I think, that is exactly what we should model as the API here. It is not necessarily Identity , but any object that has a method `getToken` that provides it with a token, which can then be used to make requests to the catalog

`getEntity(entityId: string) { let token = this.tokenProvider.getToken(); 
request(catalog-backend, token. entity)`